extensions/gdx-jnigen/src/com/badlogic/gdx/jnigen/NativeCodeGenerator.java

```diff
@@ -297,14 +301,41 @@ private void generateHFile (FileDescriptor file) throws Exception {
 	protected void emitHeaderInclude (StringBuffer buffer, String fileName) {
 		buffer.append("#include <" + fileName + ">\n");
 	}
+	
+	/** Emits a definition for desktop, so critical natives are only included in natives for desktop platforms */
+	private void emitDesktopDefinition (StringBuffer buffer) {

```

Comment:
To be honest, I don't really like this block of definitions polluting every code generated. I'd prefer the alternative idea of the first PR, like having __LIBGDX_[_]DESKTOP__ defined by the initiator of the compilation.